src/screens/FeedScreen/hooks/useFeedPreloadHook.ts

```diff
@@ -0,0 +1,26 @@
+import {NativeScrollEvent, NativeSyntheticEvent} from 'react-native';
+
+import useFeedUtilsHook from './useFeedUtilsHook';
+import {FEEDS_DIFF_TO_FETCH} from '../../../utils/constants';
+
+const useFeedPreloadHook = (feedLength: number, fetchCallback: () => void) => {
+  const {getCurrentPostViewed} = useFeedUtilsHook();
+
+  function __shouldFetchNextFeeds(momentumEvent: NativeSyntheticEvent<NativeScrollEvent>): boolean {
+    const shownIndex = getCurrentPostViewed(momentumEvent);
+    return feedLength - shownIndex <= FEEDS_DIFF_TO_FETCH;
+  }
+
+  function fetchNextFeeds(momentumEvent: NativeSyntheticEvent<NativeScrollEvent>) {
+    const shouldFetch = __shouldFetchNextFeeds(momentumEvent);
+    if (shouldFetch && fetchCallback) {
+      fetchCallback();
+    }
+  }
+
+  return {
+    fetchNextFeeds
+  };
+};
+
+export default useFeedPreloadHook;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new hook `useFeedPreloadHook` seems to be well implemented. It uses the `useFeedUtilsHook` to get the current post viewed and decides whether to fetch the next set of feeds based on the scroll event. The function `fetchNextFeeds` is correctly checking if it should fetch more feeds and if the `fetchCallback` is provided before calling it. However, there's no error handling in case `fetchCallback` fails. Consider adding a try-catch block around `fetchCallback()` to handle any potential errors.
```diff
  function fetchNextFeeds(momentumEvent: NativeSyntheticEvent<NativeScrollEvent>) {
    const shouldFetch = __shouldFetchNextFeeds(momentumEvent);
    if (shouldFetch && fetchCallback) {
      try {
        fetchCallback();
      } catch (error) {
        console.error('Failed to fetch next feeds:', error);
      }
    }
  }
```
This will ensure that any errors during the fetching process are caught and logged, improving the robustness of your code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
suggetion ini bagus, di wrap dengan try catch mas @fajarism